src/handlers/gestures/simpleGestures.ts

```diff
@@ -0,0 +1,381 @@
+    }
+  }
+
+  build(): BuiltGesture {
+    let result = new BuiltGesture(this.prepare);
+
+    result.gestures = [this];
+
+    return result;
+  }
+
+  prepare = () => {
+    if (
+      this.config.requireToFail &&
+      !Array.isArray(this.config.requireToFail)
+    ) {
+      this.config.requireToFail = [this.config.requireToFail];
+    }
+
+    if (this.config.after && !Array.isArray(this.config.after)) {
+      this.config.after = [this.config.after];
+    }
+
+    if (
+      this.config.simultaneousWith &&
+      !Array.isArray(this.config.simultaneousWith)
+    ) {
+      this.config.simultaneousWith = [this.config.simultaneousWith];
+    }
+  };
+
+  getAllowedProps(): any {
+    return SimpleGesture.allowedProps;
+  }
+
+  transformProps() {
+    return this.config;
+  }
+}
+
+export class Tap extends SimpleGesture {
+  static allowedProps = [
+    ...baseGestureHandlerWithMonitorProps,
+    ...tapGestureHandlerProps,
+  ];
+
+  constructor() {
+    super();
+
+    this.handlerName = 'TapGestureHandler';
+  }
+
+  setTapCount(count: number) {
+    this.setConfig('numberOfTaps', count);
+    return this;
+  }
+
+  setMaxDistance(maxDist: number) {
+    this.setConfig('maxDist', maxDist);
+    return this;
+  }
+
+  setMaxDuration(duration: number) {
+    this.setConfig('maxDurationMs', duration);
+    return this;
+  }
+
+  setMaxDelay(delay: number) {
+    this.setConfig('maxDelayMs', delay);
+    return this;
+  }
+
+  setMaxDeltaX(delta: number) {
+    this.setConfig('maxDeltaX', delta);
+    return this;
+  }
+
+  setMaxDeltaY(delta: number) {
+    this.setConfig('maxDeltaY', delta);
+    return this;
+  }
+
+  getAllowedProps() {
+    return Tap.allowedProps;
+  }
+}
+
+export class Pan extends SimpleGesture {
+  static allowedProps = [
+    ...baseGestureHandlerWithMonitorProps,
+    ...panGestureHandlerProps,
+    ...panGestureHandlerCustomNativeProps,
+  ];
+
+  constructor() {
+    super();
+
+    this.handlerName = 'PanGestureHandler';
+  }
+
+  setActiveOffsetY(offset: number | Array<number>) {
+    this.setConfig('activeOffsetY', offset);
+    return this;
+  }
+
+  setActiveOffsetX(offset: number | Array<number>) {
+    this.setConfig('activeOffsetX', offset);
+    return this;
+  }
+
+  setFailOffsetY(offset: number | Array<number>) {
+    this.setConfig('failOffsetY', offset);
+    return this;
+  }
+
+  setFailOffsetX(offset: number | Array<number>) {
+    this.setConfig('failOffsetX', offset);
+    return this;
+  }
+
+  setMinDistance(distance: number) {
+    this.setConfig('minDist', distance);
+    return this;
+  }
+
+  setAverageTouches(value: number) {
+    this.setConfig('avgTouches', value);
+    return this;
+  }
+
+  setEnableTrackpadTwoFingerGesture(value: boolean) {
+    this.setConfig('enableTrackpadTwoFingerGesture', value);
+    return this;
+  }
+
+  getAllowedProps() {
+    return Pan.allowedProps;
+  }
+
+  transformProps() {
+    return managePanProps(this.config);
+  }
```

Comment:
This shouldn't be part of the API.

Reply:
I removed `transformProps` and `allowedProps` in c207736e230eaea9e6e0261b51b19e1c411c7214. At the moment, only `Pan` gesture was making use of them, and this way we can update how the config is managed without sharing code with the old one.